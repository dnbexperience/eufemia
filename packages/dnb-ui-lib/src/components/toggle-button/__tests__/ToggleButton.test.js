/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  axeComponent,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../ToggleButton'

// just to make sure we re-run the test in watch mode due to changes in theese files
import _toggle_button from '../style/_toggle-button.scss' // eslint-disable-line
import dnb_toggle_button from '../style/dnb-toggle-button.scss' // eslint-disable-line
import dnb_toggle_button_theme_ui from '../style/themes/dnb-toggle-button-theme-ui.scss' // eslint-disable-line

const props = fakeProps(require.resolve('../ToggleButton'), {
  optional: true
})
props.id = 'toggle-button'
props.status = null
props.icon_position = 'left'
props.variant = 'checkbox'
props.readOnly = false

describe('ToggleButton component', () => {
  // then test the state management
  const Comp = mount(<Component {...props} />)

  // mount compare the snapshot
  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('has correct state after "click" trigger', () => {
    // default checked value has to be false
    expect(Comp.state().checked).toBe(false)

    Comp.find('button').simulate('click') // we could send inn the event data structure like this: , { target: { checked: true } }
    expect(Comp.state().checked).toBe(true)

    Comp.find('button').simulate('click')
    expect(Comp.state().checked).toBe(false)

    // also check if getDerivedStateFromProps sets the state as expected
    Comp.setProps({ checked: true })
    expect(Comp.state().checked).toBe(true)
  })

  it('has "on_change" event witch will trigger on a button click', () => {
    const my_event = jest.fn()
    const myEvent = jest.fn()
    const Comp = mount(
      <Component on_change={my_event} onChange={myEvent} checked={false} />
    )
    Comp.find('button').simulate('click')
    expect(my_event.mock.calls.length).toBe(1)
    expect(myEvent.mock.calls.length).toBe(1)
    expect(myEvent.mock.calls[0][0]).toHaveProperty('checked')
    expect(myEvent.mock.calls[0][0].checked).toBe(true)
    expect(my_event.mock.calls[0][0].checked).toBe(true)
    Comp.find('button').simulate('click')
    expect(my_event.mock.calls[1][0].checked).toBe(false)
  })

  it('has a disabled attribute, once we set disabled to true', () => {
    const Comp = mount(<Component />)
    Comp.setProps({
      disabled: true
    })
    expect(
      Comp.find('button')
        .instance()
        .hasAttribute('disabled')
    ).toBe(true)
  })

  it('should validate with ARIA rules', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})

describe('ToggleButton group component', () => {
  // then test the state management
  const Comp = mount(
    <Component.Group label="Label" id="group">
      <Component id="toggle-button-1" text="ToggleButton 1" />
      <Component id="toggle-button-2" text="ToggleButton 2" checked />
    </Component.Group>
  )

  it('has to have correct aria-pressed', () => {
    expect(
      Comp.find('button#toggle-button-2')
        .instance()
        .hasAttribute('aria-pressed')
    ).toBe(true)
  })

  it('has "on_change" event witch will trigger on a button click', () => {
    const my_event = jest.fn()
    const myEvent = jest.fn()
    const Comp = mount(
      <Component.Group
        id="group"
        on_change={my_event}
        onChange={myEvent}
        value="second"
      >
        <Component
          id="toggle-button-1"
          text="ToggleButton 1"
          value="first"
        />
        <Component
          id="toggle-button-2"
          text="ToggleButton 2"
          value="second"
        />
      </Component.Group>
    )
    Comp.find('button#toggle-button-1').simulate('click')
    expect(my_event.mock.calls.length).toBe(1)
    expect(myEvent.mock.calls.length).toBe(1)
    expect(myEvent.mock.calls[0][0]).toHaveProperty('value')
    expect(myEvent.mock.calls[0][0].value).toBe('first')
    expect(my_event.mock.calls[0][0].value).toBe('first')
    Comp.find('button#toggle-button-2').simulate('click')
    expect(my_event.mock.calls[1][0].value).toBe('second')
  })

  it('has multiselect "on_change" event witch will trigger on a button click', () => {
    const my_event = jest.fn()
    const myEvent = jest.fn()
    const Comp = mount(
      <Component.Group
        id="group"
        on_change={my_event}
        onChange={myEvent}
        values={['second']}
        multiselect="true"
      >
        <Component
          id="toggle-button-1"
          text="ToggleButton 1"
          value="first"
        />
        <Component
          id="toggle-button-2"
          text="ToggleButton 2"
          value="second"
        />
      </Component.Group>
    )
    Comp.find('button#toggle-button-1').simulate('click')
    expect(my_event.mock.calls.length).toBe(1)
    expect(myEvent.mock.calls.length).toBe(1)
    expect(myEvent.mock.calls[0][0]).toHaveProperty('values')
    expect(myEvent.mock.calls[0][0].values).toEqual(['second', 'first'])
    expect(my_event.mock.calls[0][0].values).toEqual(['second', 'first'])
    Comp.find('button#toggle-button-1').simulate('click')
    expect(my_event.mock.calls[1][0].values).toEqual(['second'])
  })

  // mount compare the snapshot
  it('have to match group snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('should validate with ARIA rules', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})

describe('ToggleButton scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/dnb-toggle-button.scss')
    )
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-toggle-button-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
