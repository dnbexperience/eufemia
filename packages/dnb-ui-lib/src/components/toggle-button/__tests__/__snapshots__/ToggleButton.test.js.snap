// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton component have to match snapshot 1`] = `
<ToggleButton
  attributes="attributes"
  checked="checked"
  class="class"
  className="className"
  custom_element={Object {}}
  custom_method={[Function]}
  direction="horizontal"
  disabled="disabled"
  group={null}
  id="id"
  label="label"
  label_position="left"
  on_change={[Function]}
  on_state_update={[Function]}
  readOnly={false}
  status={null}
  status_animation="status_animation"
  status_state="status_state"
  title="title"
  value="value"
>
  <span
    className="dnb-toggle-button dnb-toggle-button--label-position-left className class"
  >
    <FormLabel
      aria-hidden={false}
      class={null}
      className={null}
      direction={null}
      disabled={false}
      for_id="id"
      id="id-label"
      render_content={null}
      text="label"
      title={null}
      vertical={null}
    >
      <label
        aria-hidden={false}
        className="dnb-form-label"
        htmlFor="id"
        id="id-label"
      >
        label
      </label>
    </FormLabel>
    <span
      className="dnb-toggle-button__shell"
    >
      <input
        aria-checked={false}
        aria-label="title"
        aria-labelledby="id-label"
        checked={false}
        className="dnb-toggle-button__input"
        direction="horizontal"
        disabled={false}
        id="id"
        name={null}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        title="title"
        type="checkbox"
        value="value"
      />
      <span
        aria-hidden={true}
        className="dnb-toggle-button__button"
      />
      <span
        className="dnb-toggle-button__focus"
      />
      <span
        className="dnb-toggle-button__dot"
      />
    </span>
  </span>
</ToggleButton>
`;

exports[`ToggleButton group component have to match group snapshot 1`] = `
<ToggleButtonGroup
  attributes={null}
  class={null}
  className={null}
  custom_element={null}
  custom_method={null}
  direction="horizontal"
  disabled={null}
  id="group"
  label="Label"
  layout_direction="row"
  name="group"
  on_change={null}
  status={null}
  status_animation={null}
  status_state="error"
  title={null}
  value={null}
  vertical={null}
>
  <span
    className="dnb-toggle-button-group dnb-toggle-button-group--row"
  >
    <FormRow
      class={null}
      className={null}
      direction="horizontal"
      disabled={null}
      id={null}
      label="Label"
      label_id="group"
      no_label={false}
      render_content={null}
      section_spacing={null}
      section_style={null}
      size={null}
      vertical={null}
    >
      <div
        className="dnb-form-row dnb-form-row--horizontal"
      >
        <FormLabel
          class={null}
          className="dnb-form-row__label"
          direction={null}
          disabled={false}
          for_id={null}
          id="group-label"
          render_content={null}
          text="Label"
          title={null}
          vertical={null}
        >
          <label
            className="dnb-form-label dnb-form-label--horizontal dnb-form-row__label"
            id="group-label"
          >
            Label
          </label>
        </FormLabel>
        <div
          className="dnb-form-row__content"
        >
          <span
            aria-labelledby="group-label"
            className="dnb-toggle-button-group__shell"
            id="group"
            role="radiogroup"
          >
            <ToggleButton
              attributes={null}
              checked={null}
              class={null}
              className={null}
              custom_element={null}
              custom_method={null}
              disabled={false}
              group={null}
              id="toggle-button-1"
              label="ToggleButton 1"
              label_position="right"
              on_change={null}
              on_state_update={null}
              readOnly={false}
              status={null}
              status_animation={null}
              status_state="error"
              title={null}
              value=""
            >
              <span
                className="dnb-toggle-button dnb-toggle-button--label-position-right"
              >
                <FormLabel
                  aria-hidden={true}
                  class={null}
                  className={null}
                  direction={null}
                  disabled={false}
                  for_id="toggle-button-1"
                  id="toggle-button-1-label"
                  render_content={null}
                  text="ToggleButton 1"
                  title={null}
                  vertical={null}
                >
                  <label
                    aria-hidden={true}
                    className="dnb-form-label dnb-form-label--horizontal"
                    htmlFor="toggle-button-1"
                    id="toggle-button-1-label"
                  >
                    ToggleButton 1
                  </label>
                </FormLabel>
                <span
                  className="dnb-toggle-button__shell"
                >
                  <input
                    aria-checked={false}
                    aria-label={null}
                    aria-labelledby="toggle-button-1-label"
                    checked={false}
                    className="dnb-toggle-button__input"
                    disabled={false}
                    id="toggle-button-1"
                    name="group"
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onMouseOut={[Function]}
                    role="radio"
                    title={null}
                    type="radio"
                    value=""
                  />
                  <span
                    aria-hidden={true}
                    className="dnb-toggle-button__button"
                  />
                  <span
                    className="dnb-toggle-button__focus"
                  />
                  <span
                    className="dnb-toggle-button__dot"
                  />
                </span>
              </span>
            </ToggleButton>
            <ToggleButton
              attributes={null}
              checked={true}
              class={null}
              className={null}
              custom_element={null}
              custom_method={null}
              disabled={false}
              group={null}
              id="toggle-button-2"
              label="ToggleButton 2"
              label_position="right"
              on_change={null}
              on_state_update={null}
              readOnly={false}
              status={null}
              status_animation={null}
              status_state="error"
              title={null}
              value=""
            >
              <span
                className="dnb-toggle-button dnb-toggle-button--label-position-right"
              >
                <FormLabel
                  aria-hidden={true}
                  class={null}
                  className={null}
                  direction={null}
                  disabled={false}
                  for_id="toggle-button-2"
                  id="toggle-button-2-label"
                  render_content={null}
                  text="ToggleButton 2"
                  title={null}
                  vertical={null}
                >
                  <label
                    aria-hidden={true}
                    className="dnb-form-label dnb-form-label--horizontal"
                    htmlFor="toggle-button-2"
                    id="toggle-button-2-label"
                  >
                    ToggleButton 2
                  </label>
                </FormLabel>
                <span
                  className="dnb-toggle-button__shell"
                >
                  <input
                    aria-checked={true}
                    aria-label={null}
                    aria-labelledby="toggle-button-2-label"
                    checked={true}
                    className="dnb-toggle-button__input"
                    disabled={false}
                    id="toggle-button-2"
                    name="group"
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onMouseOut={[Function]}
                    role="radio"
                    title={null}
                    type="radio"
                    value=""
                  />
                  <span
                    aria-hidden={true}
                    className="dnb-toggle-button__button"
                  />
                  <span
                    className="dnb-toggle-button__focus"
                  />
                  <span
                    className="dnb-toggle-button__dot"
                  />
                </span>
              </span>
            </ToggleButton>
          </span>
        </div>
      </div>
    </FormRow>
  </span>
</ToggleButtonGroup>
`;

exports[`ToggleButton scss have to match default theme snapshot 1`] = `
"/*
* ToggleButton theme
*
*/
/**
 * This file is only used to make themes independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
.dnb-toggle-button {
  /*
    * When is OFF
    * aka when the toggle-button is not :checked
    */
  /*
    * When is ON
    * aka when the toggle-button is :checked
    */
  /*
  * When toggle-button is OFF
  * aka when the toggle-button is not :checked
  */
  /*
    * On disabled
    *
    */
  /*
    * On hover
    *
    */
  /*
    * On active
    *
    */
  /* stylelint-disable */
  /* stylelint-enable */
  /*
    * On focus
    *
    */
  /* stylelint-disable */
  /* stylelint-enable */
  /*
    * On error state
    *
    */
  /* stylelint-disable */
  /* stylelint-enable */ }
  .dnb-toggle-button__dot {
    background-color: var(--color-sea-green-alt);
    transition: opacity 200ms ease-out, transform 200ms ease-out; }
  .dnb-toggle-button__input:not(:checked) ~ .dnb-toggle-button__button {
    background-color: var(--color-white);
    border-color: var(--color-sea-green-alt); }
  .dnb-toggle-button__input:checked ~ .dnb-toggle-button__button {
    background-color: var(--color-white);
    border-color: var(--color-sea-green-alt); }
  .dnb-toggle-button__input:checked ~ .dnb-toggle-button__dot {
    opacity: 1;
    transform: scale(1); }
  .dnb-toggle-button__input:not(:checked) ~ .dnb-toggle-button__dot {
    opacity: 0;
    transform: scale(0.8); }
  .dnb-toggle-button__input[disabled] ~ .dnb-toggle-button__button {
    border-color: var(--color-mint-green-50); }
  .dnb-toggle-button__input[disabled]:checked ~ .dnb-toggle-button__button {
    border-color: var(--color-sea-green-alt-30); }
  .dnb-toggle-button__input[disabled]:checked ~ .dnb-toggle-button__dot {
    background-color: var(--color-sea-green-alt-30); }
  .dnb-toggle-button__input:not([disabled]):not(:focus):not(:checked):hover ~ .dnb-toggle-button__button {
    background-color: var(--color-mint-green-50); }
  .dnb-toggle-button__input:not([disabled]):not(:focus):checked:hover ~ .dnb-toggle-button__dot {
    background-color: var(--color-mint-green-50); }
  .dnb-toggle-button__input:not([disabled]):not(:checked):active ~ .dnb-toggle-button__button {
    background-color: var(--color-mint-green-50);
    border-color: transparent; }
  .dnb-toggle-button__input:not([disabled]):checked:active ~ .dnb-toggle-button__button {
    border-color: var(--color-mint-green-50); }
  .dnb-toggle-button__input:not([disabled]):checked:active ~ .dnb-toggle-button__dot {
    background-color: var(--color-mint-green-50); }
  html[data-whatinput='keyboard'] .dnb-toggle-button__input:not([disabled]):not(:checked):focus ~ .dnb-toggle-button__button {
    background-color: var(--color-mint-green-50); }
  html[data-whatinput='keyboard'] .dnb-toggle-button__input:not([disabled]):focus ~ .dnb-toggle-button__button {
    border: none; }
  .dnb-toggle-button__input:not([disabled]):focus ~ .dnb-toggle-button__focus,
  .dnb-toggle-button__input:not([disabled]):active ~ .dnb-toggle-button__focus {
    display: block; }
  .dnb-toggle-button__input:not([disabled]):not(:checked):focus ~ .dnb-toggle-button__button,
  .dnb-toggle-button__input:not([disabled]):not(:checked):active ~ .dnb-toggle-button__button {
    background-color: var(--color-mint-green-50); }
  .dnb-toggle-button__status--error .dnb-toggle-button__input:not([disabled]):not(:focus):not(:active) ~ .dnb-toggle-button__button {
    border: none; }
  .dnb-toggle-button__status--error .dnb-toggle-button__input:not([disabled]):not(:focus):not(:active) ~ .dnb-toggle-button__focus {
    display: block;
    box-shadow: 0 0 0 0.125rem var(--color-cherry-red);
    border-color: transparent; }
  .dnb-toggle-button__status--error .dnb-toggle-button__input:not([disabled]):not(:focus):not(:checked):hover ~ .dnb-toggle-button__button {
    background-color: var(--color-cherry-red-80); }
  .dnb-toggle-button__status--error .dnb-toggle-button__input:not([disabled]):not(:focus):not(:active):not(:hover):checked ~ .dnb-toggle-button__dot {
    background-color: var(--color-cherry-red); }
  .dnb-toggle-button__status--error .dnb-toggle-button__input:not([disabled]):not(:focus):not(:active):checked:hover ~ .dnb-toggle-button__dot {
    background-color: var(--color-cherry-red-80); }
  .dnb-toggle-button > .dnb-form-status {
    transform: translateY(0.1875rem); }
"
`;

exports[`ToggleButton scss have to match snapshot 1`] = `
"/*
* DNB ToggleButton
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * Scopes
 *
 */
/*
 * Document Reset
 *
 */
/*
* DNB FormLabel
*
*/
.dnb-form-label {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: 1rem;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333);
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\";
  /**
  * 1. Remove repeating backgrounds in all browsers (opinionated).
  * 2. Add border box sizing in all browsers (opinionated).
  */
  /**
  * 1. Add text decoration inheritance in all browsers (opinionated).
  * 2. Add vertical alignment inheritance in all browsers (opinionated).
  */
  margin: 0;
  padding: 0; }
  .dnb-form-label *,
  .dnb-form-label ::before,
  .dnb-form-label ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-form-label ::before,
  .dnb-form-label ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }

/*
 * FormLabel component
 *
 */
.dnb-form-label {
  display: inline-block;
  width: auto;
  margin: 0;
  margin-right: 1rem;
  vertical-align: baseline;
  color: inherit; }
  .dnb-form-label--vertical {
    display: block;
    margin-right: 0;
    margin-bottom: 0.5rem; }
  .dnb-form-label[for]:not([disabled]) {
    user-select: none;
    cursor: pointer; }
  .dnb-form-label[disabled] {
    cursor: not-allowed; }

/*
* DNB FormStatus
*
*/
.dnb-form-status {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: 1rem;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333);
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\";
  /**
  * 1. Remove repeating backgrounds in all browsers (opinionated).
  * 2. Add border box sizing in all browsers (opinionated).
  */
  /**
  * 1. Add text decoration inheritance in all browsers (opinionated).
  * 2. Add vertical alignment inheritance in all browsers (opinionated).
  */
  margin: 0;
  padding: 0; }
  .dnb-form-status *,
  .dnb-form-status ::before,
  .dnb-form-status ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-form-status ::before,
  .dnb-form-status ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }

/*
 * FormStatus component
 *
 */
.dnb-form-status {
  width: auto;
  display: flex;
  min-height: var(--input-height); }
  .dnb-form-status__shell {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    border-radius: var(--input-border-radius); }
  .dnb-form-status--text {
    padding: calc(0.5rem / 2) 1rem;
    cursor: text;
    color: var(--color-emerald-green);
    line-height: 1.5rem; }
  .dnb-icon + .dnb-form-status--text {
    padding-left: 0.5rem; }
  .dnb-form-status__shell > .dnb-icon {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 1.5em;
    height: 1.5em;
    margin-left: 0.5rem;
    color: inherit;
    font-size: 1rem;
    border-radius: 50%;
    border: 1px solid;
    border-color: currentColor; }
  .dnb-form-status[hidden] {
    display: none; }
  .dnb-form-status--fade-in {
    height: 0;
    opacity: 0;
    animation: fade-in 600ms ease-out 1 200ms forwards; }

@keyframes fade-in {
  from {
    opacity: 0;
    height: 0; }
  to {
    opacity: 1;
    height: var(--input-height); } }

.dnb-toggle-button {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: 1rem;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333);
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\";
  /**
  * 1. Remove repeating backgrounds in all browsers (opinionated).
  * 2. Add border box sizing in all browsers (opinionated).
  */
  /**
  * 1. Add text decoration inheritance in all browsers (opinionated).
  * 2. Add vertical alignment inheritance in all browsers (opinionated).
  */
  margin: 0;
  padding: 0; }
  .dnb-toggle-button *,
  .dnb-toggle-button ::before,
  .dnb-toggle-button ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-toggle-button ::before,
  .dnb-toggle-button ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }

/*
* ToggleButton component
*
*/
:root {
  --toggle-button-height: 1.5rem;
  --toggle-button-width: 1.5rem;
  --toggle-button-border-width: 0.125rem; }

.dnb-toggle-button {
  display: inline-flex;
  flex-direction: row;
  width: auto;
  margin: 0;
  padding: 0; }
  .dnb-toggle-button__shell {
    position: relative;
    user-select: none;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: var(--toggle-button-height);
    height: var(--toggle-button-height);
    transform: translateY(-0.0625rem); }
  .dnb-toggle-button__focus, .dnb-toggle-button__button, .dnb-toggle-button__dot {
    position: absolute;
    z-index: 4; }
  .dnb-toggle-button__button {
    border: var(--toggle-button-border-width) solid transparent; }
  .dnb-toggle-button__focus {
    display: none;
    outline: none; }
    html[data-whatinput='keyboard'] .dnb-toggle-button__focus {
      box-shadow: 0 0 0 0.125rem var(--color-emerald-green);
      border-color: transparent; }
  .dnb-toggle-button__focus, .dnb-toggle-button__button {
    width: calc(var(--toggle-button-width) - 0.25rem);
    height: calc(var(--toggle-button-height) - 0.25rem);
    border-radius: 50%; }
  .dnb-toggle-button__dot {
    width: calc(var(--toggle-button-width) - 0.75rem);
    height: calc(var(--toggle-button-height) - 0.75rem);
    border-radius: 50%; }
  .dnb-toggle-button__input {
    opacity: 0;
    position: absolute;
    left: 0;
    z-index: 5;
    width: var(--toggle-button-width);
    height: var(--toggle-button-height);
    margin: 0;
    padding: 0;
    border: 0; }
  .dnb-toggle-button__input:not([disabled]) {
    cursor: pointer; }
  .dnb-toggle-button .dnb-form-status {
    margin-bottom: 0.5rem; }
  .dnb-toggle-button--modifier {
    display: inline-flex;
    flex-direction: row; }
  .dnb-toggle-button--label-position-left .dnb-toggle-button {
    order: 2; }
  .dnb-toggle-button--label-position-left .dnb-form-label {
    order: 1;
    margin-right: 0.5rem;
    margin-left: 1rem; }
  .dnb-toggle-button-group--column .dnb-toggle-button--label-position-left {
    display: flex;
    margin-bottom: 0.5rem; }
  .dnb-toggle-button--label-position-right .dnb-toggle-button {
    order: 1; }
  .dnb-toggle-button--label-position-right .dnb-form-label {
    order: 2;
    margin-right: 1rem;
    margin-left: 0.5rem; }
  .dnb-toggle-button-group--column .dnb-toggle-button--label-position-right {
    display: flex;
    margin-bottom: 0.5rem; }
  :not(.dnb-toggle-button-group) > .dnb-form-label + .dnb-toggle-button {
    vertical-align: top; }
  .dnb-toggle-button--label-position-left ~ .dnb-form-status,
  .dnb-toggle-button--label-position-right ~ .dnb-form-status {
    margin-bottom: 1rem; }
  .dnb-toggle-button-group {
    display: inline-flex;
    flex-direction: row; }
"
`;
